include/swift/AST/Decl.h

```diff
@@ -4915,6 +4908,9 @@ class VarDecl : public AbstractStorageDecl {
     Parent = v;
   }
 
+  NamedPattern *getNamingPattern() const { return NamingPattern; }
```

Comment:
The naming pattern is a compromise and a cache.  We could absolutely requestify this and have it computed by rooting around in the parent pattern binding for the NamedPattern that has us/our canonical var as its decl.